### PR TITLE
KvMemNN for ConvAI2 +  download & seq2seq fixes

### PR DESCRIPTION
**Pull request by jaseweston**

### CODE REVIEW

File: parlai/agents/seq2seq/seq2seq.py [hunk 1]
```diff
@@ -593,7 +593,10 @@ def shutdown(self):
     def load(self, path):
         """Return opt and model states."""
         with open(path, 'rb') as read:
-            states = torch.load(read, map_location='cpu')
+            try:
+                states = torch.load(read, map_location='cpu')
+            except:
```
Comment (alexholdenmiller) [inline parlai/agents/seq2seq/seq2seq.py]: you should except the exact error (`TypeError`)

File: projects/convai2/baselines/kvmemnn/eval.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+'''Evaluate pre-trained model trained for hits@1 metric
+Key-Value Memory Net model trained on personachat using persona 'self'
+'''
+
+if __name__ == '__main__':
+    parser = ParlaiParser(add_model_args=True)
+    parser.add_argument('-n', '--num-examples', default=100000000)
+    parser.add_argument('-d', '--display-examples', type='bool', default=False)
+    parser.add_argument('-ltim', '--log-every-n-secs', type=float, default=2)
```
Comment (alexholdenmiller) [inline projects/convai2/baselines/kvmemnn/eval.py]: you can replace these four lines with `setup_args` from eval_model, it was just duplicating that work

File: projects/convai2/baselines/seq2seq/#eval.py# [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        model='seq2seq',
+        model_file='models:convai2/seq2seq/convai2_self_seq2seq_model',
+        dict_file='models:convai2/seq2seq/dict_convai2_self/dict_convai2_self',
+        datatype='valid',
+        batchsize=128,
+    )
+    opt = parser.parse_args()
+    download(opt, 'convai2/seq2seq', 'convai2_self_seq2seq_model.tgz')
+    download(opt, 'convai2/seq2seq', 'dict_convai2_self')
+    eval_model(parser, printargs=False)
```
Comment (alexholdenmiller) [inline projects/convai2/baselines/seq2seq/#eval.py#]: oops

**Comment by jaseweston** [comment]
cleaned

**Pull request merged**